src/main/scala/tile/FPU.scala

```diff
@@ -221,132 +202,222 @@ object ClassifyRecFN {
   }
 }
 
-object IsNaNRecFN {
-  def apply(in: UInt, t: FType) = in(t.sig + t.exp - 1, t.sig + t.exp - 3).andR
-}
-
-object IsSNaNRecFN {
-  def apply(in: UInt, t: FType) = IsNaNRecFN(in, t) && !in(t.sig - 2)
-}
-
-class FType(val exp: Int, val sig: Int)
+case class FType(exp: Int, sig: Int) {
+  def ieeeWidth = exp + sig
+  def recodedWidth = ieeeWidth + 1
 
-object FType {
-  val S = new FType(8, 24)
-  val D = new FType(11, 53)
-}
+  def qNaN = UInt((BigInt(7) << (exp + sig - 3)) + (BigInt(1) << (sig - 2)), exp + sig + 1)
+  def isNaN(x: UInt) = x(sig + exp - 1, sig + exp - 3).andR
+  def isSNaN(x: UInt) = isNaN(x) && !x(sig - 2)
 
-/** Format conversion without rounding or NaN handling */
-object RecFNToRecFN_noncompliant {
-  def apply(in: UInt, inExpWidth: Int, inSigWidth: Int, outExpWidth: Int, outSigWidth: Int) = {
-    val sign = in(inSigWidth + inExpWidth)
-    val fractIn = in(inSigWidth - 2, 0)
-    val expIn = in(inSigWidth + inExpWidth - 1, inSigWidth - 1)
-    val fractOut = fractIn << outSigWidth >> inSigWidth
+  // convert between formats, ignoring rounding, range, NaN
+	def unsafeConvert(x: UInt, to: FType) = if (this == to) x else {
```

Comment:
Yeah, for some reason a tab ended up in there?  Not sure how, since I have expandtab on in vim... but I didn't see it, since I have ts=2.